### PR TITLE
Adds default tags support to AWS Glue resources

### PR DESCRIPTION
**Pull request by gdavison**
Adds default tags support for AWS Glue resources.

Relates #7926

### CODE REVIEW

File: website/docs/r/glue_crawler.html.markdown [hunk 1]
```diff
@@ -149,7 +149,8 @@ The following arguments are supported:
 * `recrawl_policy` (Optional)  A policy that specifies whether to crawl the entire dataset again, or to crawl only folders that were added since the last crawler run.. See [Recrawl Policy](#recrawl-policy) below.
 * `security_configuration` (Optional) The name of Security Configuration to be used by the crawler
 * `table_prefix` (Optional) The table prefix used for catalog tables that are created.
-* `tags` - (Optional) Key-value map of resource tags
+* `tags` - (Optional) Key-value map of resource tags. If configured with a provider [`default_tags` configuration block](/docs/providers/aws/index.html#default_tags-configuration-block) present, tags with matching keys will overwrite those defined at the provider-level.
+* `tags_all` - (Optional) Key-value map of resource tags, including those inherited from the provider [`default_tags` configuration block](/docs/providers/aws/index.html#default_tags-configuration-block).
```
Comment (anGie44) [inline website/docs/r/glue_crawler.html.markdown:153]: `tags_all` should be under the `Attributes` heading

File: website/docs/r/glue_dev_endpoint.markdown [hunk 2]
```diff
@@ -59,7 +59,8 @@ The following arguments are supported:
 * `security_configuration` - (Optional) The name of the Security Configuration structure to be used with this endpoint.
 * `security_group_ids` - (Optional) Security group IDs for the security groups to be used by this endpoint.
 * `subnet_id` - (Optional) The subnet ID for the new endpoint to use.
-* `tags` - (Optional) Key-value map of resource tags.
+* `tags` - (Optional) Key-value map of resource tags. If configured with a provider [`default_tags` configuration block](/docs/providers/aws/index.html#default_tags-configuration-block) present, tags with matching keys will overwrite those defined at the provider-level.
+* `tags_all` - (Optional) Key-value map of resource tags, including those inherited from the provider [`default_tags` configuration block](/docs/providers/aws/index.html#default_tags-configuration-block).
```
Comment (anGie44) [inline website/docs/r/glue_dev_endpoint.markdown:63]: similar comment as above

File: website/docs/r/glue_job.html.markdown [hunk 3]
```diff
@@ -81,7 +81,8 @@ The following arguments are supported:
 * `name` – (Required) The name you assign to this job. It must be unique in your account.
 * `notification_property` - (Optional) Notification property of the job. Defined below.
 * `role_arn` – (Required) The ARN of the IAM role associated with this job.
-* `tags` - (Optional) Key-value map of resource tags
+* `tags` - (Optional) Key-value map of resource tags. If configured with a provider [`default_tags` configuration block](/docs/providers/aws/index.html#default_tags-configuration-block) present, tags with matching keys will overwrite those defined at the provider-level.
+* `tags_all` - (Optional) Key-value map of resource tags, including those inherited from the provider [`default_tags` configuration block](/docs/providers/aws/index.html#default_tags-configuration-block).
```
Comment (anGie44) [inline website/docs/r/glue_job.html.markdown:85]: similar comment as above

File: website/docs/r/glue_ml_transform.html.markdown [hunk 4]
```diff
@@ -132,7 +132,8 @@ The following arguments are supported:
 * `glue_version` - (Optional) The version of glue to use, for example "1.0". For information about available versions, see the [AWS Glue Release Notes](https://docs.aws.amazon.com/glue/latest/dg/release-notes.html).
 * `max_capacity` – (Optional) The number of AWS Glue data processing units (DPUs) that are allocated to task runs for this transform. You can allocate from `2` to `100` DPUs; the default is `10`. `max_capacity` is a mutually exclusive option with `number_of_workers` and `worker_type`.
 * `max_retries` – (Optional) The maximum number of times to retry this ML Transform if it fails.
-* `tags` - (Optional) Key-value map of resource tags
+* `tags` - (Optional) Key-value map of resource tags. If configured with a provider [`default_tags` configuration block](/docs/providers/aws/index.html#default_tags-configuration-block) present, tags with matching keys will overwrite those defined at the provider-level.
+* `tags_all` - (Optional) Key-value map of resource tags, including those inherited from the provider [`default_tags` configuration block](/docs/providers/aws/index.html#default_tags-configuration-block).
```
Comment (anGie44) [inline website/docs/r/glue_ml_transform.html.markdown:136]: similar comment as above

File: website/docs/r/glue_registry.html.markdown [hunk 5]
```diff
@@ -24,7 +24,8 @@ The following arguments are supported:
 
 * `registry_name` – (Required) The Name of the registry.
 * `description` – (Optional) A description of the registry.
-* `tags` - (Optional) Key-value map of resource tags
+* `tags` - (Optional) Key-value map of resource tags. If configured with a provider [`default_tags` configuration block](/docs/providers/aws/index.html#default_tags-configuration-block) present, tags with matching keys will overwrite those defined at the provider-level.
+* `tags_all` - (Optional) Key-value map of resource tags, including those inherited from the provider [`default_tags` configuration block](/docs/providers/aws/index.html#default_tags-configuration-block).
```
Comment (anGie44) [inline website/docs/r/glue_registry.html.markdown:28]: similar comment as above

File: website/docs/r/glue_schema.html.markdown [hunk 6]
```diff
@@ -32,7 +32,8 @@ The following arguments are supported:
 * `compatibility` - (Required) The compatibility mode of the schema. Values values are: `NONE`, `DISABLED`, `BACKWARD`, `BACKWARD_ALL`, `FORWARD`, `FORWARD_ALL`, `FULL`, and `FULL_ALL`.
 * `schema_definition` - (Required) The schema definition using the `data_format` setting for `schema_name`.
 * `description` – (Optional) A description of the schema.
-* `tags` - (Optional) Key-value map of resource tags
+* `tags` - (Optional) Key-value map of resource tags. If configured with a provider [`default_tags` configuration block](/docs/providers/aws/index.html#default_tags-configuration-block) present, tags with matching keys will overwrite those defined at the provider-level.
+* `tags_all` - (Optional) Key-value map of resource tags, including those inherited from the provider [`default_tags` configuration block](/docs/providers/aws/index.html#default_tags-configuration-block).
```
Comment (anGie44) [inline website/docs/r/glue_schema.html.markdown:36]: similar comment as above

File: website/docs/r/glue_trigger.html.markdown [hunk 7]
```diff
@@ -113,7 +113,8 @@ The following arguments are supported:
 * `name` – (Required) The name of the trigger.
 * `predicate` – (Optional) A predicate to specify when the new trigger should fire. Required when trigger type is `CONDITIONAL`. See [Predicate](#predicate) Below.
 * `schedule` – (Optional) A cron expression used to specify the schedule. [Time-Based Schedules for Jobs and Crawlers](https://docs.aws.amazon.com/glue/latest/dg/monitor-data-warehouse-schedule.html)
-* `tags` - (Optional) Key-value map of resource tags
+* `tags` - (Optional) Key-value map of resource tags. If configured with a provider [`default_tags` configuration block](/docs/providers/aws/index.html#default_tags-configuration-block) present, tags with matching keys will overwrite those defined at the provider-level.
+* `tags_all` - (Optional) Key-value map of resource tags, including those inherited from the provider [`default_tags` configuration block](/docs/providers/aws/index.html#default_tags-configuration-block).
```
Comment (anGie44) [inline website/docs/r/glue_trigger.html.markdown:117]: similar comment as above

File: website/docs/r/glue_workflow.html.markdown [hunk 8]
```diff
@@ -55,7 +55,8 @@ The following arguments are supported:
 * `default_run_properties` – (Optional) A map of default run properties for this workflow. These properties are passed to all jobs associated to the workflow.
 * `description` – (Optional) Description of the workflow.
 * `max_concurrent_runs` - (Optional) Prevents exceeding the maximum number of concurrent runs of any of the component jobs. If you leave this parameter blank, there is no limit to the number of concurrent workflow runs.
-* `tags` - (Optional) Key-value map of resource tags
+* `tags` - (Optional) Key-value map of resource tags. If configured with a provider [`default_tags` configuration block](/docs/providers/aws/index.html#default_tags-configuration-block) present, tags with matching keys will overwrite those defined at the provider-level.
+* `tags_all` - (Optional) Key-value map of resource tags, including those inherited from the provider [`default_tags` configuration block](/docs/providers/aws/index.html#default_tags-configuration-block).
```
Comment (anGie44) [inline website/docs/r/glue_workflow.html.markdown:59]: similar comment as above

**Review by anGie44 — APPROVED** [review]
Just a couple documentation edits needed, otherwise LGTM

File: website/docs/r/glue_crawler.html.markdown (hunk 1 above)
Comment (gdavison) [reply website/docs/r/glue_crawler.html.markdown:153]: 🤦

**Review by anGie44 — APPROVED** [review]
Looks great 🚀 !

**Pull request merged**